### PR TITLE
[text_splitter]: Added Semantic text splitter for better performance and more option of splitter to user

### PR DESCRIPTION
**Pull request by rgupta2508**
text_splitter: Added Semantic text splitter for better performance and more option of splitter to user
Adding SemanticCharacterTextSplitter, SemanticTiktokenTextSplitter

Additional guidelines:
- Make sure optional dependencies are imported within a function.
- Please do not add dependencies to pyproject.toml files (even optional ones) unless they are required for unit tests.
- Most PRs should not touch more than one package.
- Changes should be backwards compatible.
- If you are adding something to community, do not re-import it in langchain.

If no one reviews your PR within a few days, please @-mention one of baskaryan, efriis, eyurtsev, hwchase17.

### CODE REVIEW

**Review by baskaryan — COMMENTED** [review]
could we add some docs that show how these work?

**Comment by rgupta2508** [comment]
> could we add some docs that show how these work?

@baskaryan Added helping docs to which explain how it works

**Review by hwchase17 — COMMENTED** [review]
lets add this to langchain-community or langchain-experimental. langchain-text-splitters is meant for text splitters we implement, while this is more of an integraiton

**Comment by rgupta2508** [comment]
> lets add this to langchain-community or langchain-experimental. langchain-text-splitters is meant for text splitters we implement, while this is more of an integraiton

@hwchase17  agree . Moved this integration to langchain-community. please review

File: libs/community/pyproject.toml [hunk 1]
```diff
@@ -97,6 +97,8 @@ rdflib = {version = "7.0.0", optional = true}
 nvidia-riva-client = {version = "^2.14.0", optional = true}
 tidb-vector = {version = ">=0.0.3,<1.0.0", optional = true}
 friendli-client = {version = "^1.2.4", optional = true}
+semantic-text-splitter = {version = "0.6.3", optional = false}
```
Comment (hwchase17) [inline libs/community/pyproject.toml:100]: this should be optional = true

File: docs/docs/modules/data_connection/document_transformers/semantic_character_text_splitter.ipynb [hunk 2]
```diff
@@ -0,0 +1,127 @@
+{
```
Comment (hwchase17) [inline docs/docs/modules/data_connection/document_transformers/semantic_character_text_splitter.ipynb:1]: i think combining these into one notebook is probably best, theyre pretty similar

Comment (rgupta2508) [reply docs/docs/modules/data_connection/document_transformers/semantic_character_text_splitter.ipynb]: > i think combining these into one notebook is probably best, theyre pretty similar

@hwchase17  Ok Yes, Combined these 2 Splitter in one docs

File: libs/community/pyproject.toml (hunk 1 above)
Comment (rgupta2508) [reply libs/community/pyproject.toml:100]: Done

File: docs/docs/modules/data_connection/document_transformers/index.mdx [hunk 3]
```diff
@@ -43,6 +43,8 @@ LangChain offers many different types of text splitters. These all live in the `
...
 | Token     | Tokens                                |               | Splits text on tokens. There exist a few different ways to measure tokens.                                                                                                              |
 | Character | A user defined character              |               | Splits text based on a user defined character. One of the simpler methods.                                                                                                              |
+| SemanticCharacter | Sentences                     |               | Splits text based on the Semantic character.                                                                                                              |
+| SemanticTiktoken | Sentences                      |               | Splits text based on the Semantic Token.                                                                                                              |
```
Comment (hwchase17) [inline docs/docs/modules/data_connection/document_transformers/index.mdx:47]:
```suggestion
| Semantic Text Splitter | Sentences                     |               | Splits text based on the `semantic_text_splitter` package.                                                                                                              |
```

**Review by hwchase17 — COMMENTED** [review]
the notebook needs work. the title needs to reflect the fact that it combines both of them - right now you have two H1s

Theres also just basic stuff that seems wrong. `[These notes](https://pypi.org/project/semantic-text-splitter/) from Pinecone provide some useful knowledge about this how it works` - why is this linking to pypi package but mentioning Pinecone? Also sorry - does this whole package operate by splitting on `\\n\\n\`? how is this different from our internal character text splitter then?

**Comment by rgupta2508** [comment]
> the notebook needs work. the title needs to reflect the fact that it combines both of them - right now you have two H1s
> 
> Theres also just basic stuff that seems wrong. `[These notes](https://pypi.org/project/semantic-text-splitter/) from Pinecone provide some useful knowledge about this how it works` - why is this linking to pypi package but mentioning Pinecone? Also sorry - does this whole package operate by splitting on `\\n\\n\`? how is this different from our internal character text splitter then?

Ok @hwchase17 , Sorry That was some copy paste mistake, i have added some more detail about this text splitter. Please review

**Comment by rgupta2508** [comment]
@hwchase17  Cant able to fix some of the tests,, can you please share logic of these tests, 

tests/unit_tests/test_sql_database_schema.py::test_table_info FAILED     [  1%]
tests/unit_tests/test_sql_database_schema.py::test_sql_database_run FAILED [  1%]


Not sure these are related to this..

**Comment by rgupta2508** [comment]
@hwchase17  Please review if this can be merged?

**Comment by rgupta2508** [comment]
@hwchase17 Can you please review if this can be merged? Resolved all the review comments.

**Comment by rgupta2508** [comment]
@baskaryan  any suggestion on this? if this can be merged?

**Comment by rgupta2508** [comment]
Hi @baskaryan  any suggestion on this? can you please review this, if this can be merged. all the review comments in incorporated and document also updated at correct place .
Thanks

**Pull request closed without merging**